packages/contracts/contracts/TransactionManager.sol

```diff
@@ -416,34 +416,42 @@ contract TransactionManager is ReentrancyGuard, ProposedOwnable, ITransactionMan
     // Get the hash of the invariant tx data. This hash is the same
     // between sending and receiving chains. The variant data is stored
     // in the contract when `prepare` is called within the mapping.
-    bytes32 digest = hashInvariantTransactionData(txData);
 
-    // Make sure that the variant data matches what was stored
-    require(variantTransactionData[digest] == hashVariantTransactionData(txData.amount, txData.expiry, txData.preparedBlockNumber), "#F:019");
+    { // scope: validation and effects
+      bytes32 digest = hashInvariantTransactionData(txData);
```

Comment:
why is this not `memory`?

Reply:
the `bytes32` doesn't need the `memory` / `calldata` defined (similar to `uint256`) because its size is known